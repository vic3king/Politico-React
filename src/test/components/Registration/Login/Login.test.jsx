import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import LoginPage from '../../../../components/Registration/Login';

const initialState = {
  auth: {
    isAdmin: false,
    isCitizen: false,
    isPolitician: false,
    redirect: false,
    token: null,
  },
};

const props = {
  login: jest.fn(),
  isLoadingReducer: { loader: true },
  auth: { isAdmin: '', isCitizen: '', isPolitician: '' },
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

const LoginPageComponent = (
  <Provider store={store}>
    <LoginPage auth={{ fake: 'test' }} {...props} />
  </Provider>
);

describe('<LoginPage />', () => {
  it('renders login page', () => {
    const wrapper = shallow(LoginPageComponent);
    expect(wrapper.find('div'));
  });
});

describe('<LoginPage />', () => {
  it('renders login page', () => {
    const inputEvent = { target: { id: 'name', value: 'pcp' } };
    const wrapper = shallow(<LoginPage auth={{ fake: 'yup' }} {...props} />);
    expect(wrapper.find('div'));
    wrapper.instance().onInputChange(inputEvent);
    wrapper.instance().onButtonSubmit();
  });
});
