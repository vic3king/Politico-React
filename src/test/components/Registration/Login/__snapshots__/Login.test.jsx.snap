// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        <NavBar>
          <LiTag
            to="/signup"
            value="Sign Up"
          />
        </NavBar>,
        <div
          className="d-flex justify-content-center"
        >
          <div
            className="inner"
          >
            <div
              className="center"
            >
              <i
                aria-hidden="true"
                className="fa fa-balance-scale fa-5x "
              />
              <h3
                className="welcome__"
              >
                Please login
              </h3>
            </div>
            <form
              className="signbox login"
              id="login-form"
              onSubmit={[Function]}
            >
              <Input
                id="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
                value=""
              />
              <span
                id="spanemail"
              />
              <Input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
              <span
                id="spanpassword"
              />
              <Input
                className="submit"
                id="submit"
                name=""
                placeholder=""
                type="submit"
                value="Login"
              />
              <Link
                className="link"
                to="/reset"
              >
                Forgotten your password?
              </Link>
              <p
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                Need an account?
                <Link
                  className="link"
                  to="/signup"
                >
                  Sign Up
                </Link>
              </p>
            </form>
          </div>
        </div>,
        false,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <LiTag
            to="/signup"
            value="Sign Up"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "to": "/signup",
            "value": "Sign Up",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner"
          >
            <div
              className="center"
            >
              <i
                aria-hidden="true"
                className="fa fa-balance-scale fa-5x "
              />
              <h3
                className="welcome__"
              >
                Please login
              </h3>
            </div>
            <form
              className="signbox login"
              id="login-form"
              onSubmit={[Function]}
            >
              <Input
                id="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
                value=""
              />
              <span
                id="spanemail"
              />
              <Input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
              <span
                id="spanpassword"
              />
              <Input
                className="submit"
                id="submit"
                name=""
                placeholder=""
                type="submit"
                value="Login"
              />
              <Link
                className="link"
                to="/reset"
              >
                Forgotten your password?
              </Link>
              <p
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                Need an account?
                <Link
                  className="link"
                  to="/signup"
                >
                  Sign Up
                </Link>
              </p>
            </form>
          </div>,
          "className": "d-flex justify-content-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="center"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-balance-scale fa-5x "
                />
                <h3
                  className="welcome__"
                >
                  Please login
                </h3>
              </div>,
              <form
                className="signbox login"
                id="login-form"
                onSubmit={[Function]}
              >
                <Input
                  id="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                  value=""
                />
                <span
                  id="spanemail"
                />
                <Input
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
                <span
                  id="spanpassword"
                />
                <Input
                  className="submit"
                  id="submit"
                  name=""
                  placeholder=""
                  type="submit"
                  value="Login"
                />
                <Link
                  className="link"
                  to="/reset"
                >
                  Forgotten your password?
                </Link>
                <p
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Need an account?
                  <Link
                    className="link"
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </p>
              </form>,
            ],
            "className": "inner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-balance-scale fa-5x "
                  />,
                  <h3
                    className="welcome__"
                  >
                    Please login
                  </h3>,
                ],
                "className": "center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-balance-scale fa-5x ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Please login",
                    "className": "welcome__",
                  },
                  "ref": null,
                  "rendered": "Please login",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Input
                    id="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                    value=""
                  />,
                  <span
                    id="spanemail"
                  />,
                  <Input
                    id="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
                  />,
                  <span
                    id="spanpassword"
                  />,
                  <Input
                    className="submit"
                    id="submit"
                    name=""
                    placeholder=""
                    type="submit"
                    value="Login"
                  />,
                  <Link
                    className="link"
                    to="/reset"
                  >
                    Forgotten your password?
                  </Link>,
                  <p
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Need an account?
                    <Link
                      className="link"
                      to="/signup"
                    >
                      Sign Up
                    </Link>
                  </p>,
                ],
                "className": "signbox login",
                "id": "login-form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "email",
                    "onChange": [Function],
                    "placeholder": "Email",
                    "required": true,
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "spanemail",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "password",
                    "onChange": [Function],
                    "placeholder": "Password",
                    "required": true,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "spanpassword",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "submit",
                    "id": "submit",
                    "name": "",
                    "placeholder": "",
                    "type": "submit",
                    "value": "Login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Forgotten your password?",
                    "className": "link",
                    "to": "/reset",
                  },
                  "ref": null,
                  "rendered": "Forgotten your password?",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Need an account?",
                      <Link
                        className="link"
                        to="/signup"
                      >
                        Sign Up
                      </Link>,
                    ],
                    "style": Object {
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    "Need an account?",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Sign Up",
                        "className": "link",
                        "to": "/signup",
                      },
                      "ref": null,
                      "rendered": "Sign Up",
                      "type": [Function],
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      false,
      false,
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          <NavBar>
            <LiTag
              to="/signup"
              value="Sign Up"
            />
          </NavBar>,
          <div
            className="d-flex justify-content-center"
          >
            <div
              className="inner"
            >
              <div
                className="center"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-balance-scale fa-5x "
                />
                <h3
                  className="welcome__"
                >
                  Please login
                </h3>
              </div>
              <form
                className="signbox login"
                id="login-form"
                onSubmit={[Function]}
              >
                <Input
                  id="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                  value=""
                />
                <span
                  id="spanemail"
                />
                <Input
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
                <span
                  id="spanpassword"
                />
                <Input
                  className="submit"
                  id="submit"
                  name=""
                  placeholder=""
                  type="submit"
                  value="Login"
                />
                <Link
                  className="link"
                  to="/reset"
                >
                  Forgotten your password?
                </Link>
                <p
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Need an account?
                  <Link
                    className="link"
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </p>
              </form>
            </div>
          </div>,
          false,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <LiTag
              to="/signup"
              value="Sign Up"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "to": "/signup",
              "value": "Sign Up",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inner"
            >
              <div
                className="center"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-balance-scale fa-5x "
                />
                <h3
                  className="welcome__"
                >
                  Please login
                </h3>
              </div>
              <form
                className="signbox login"
                id="login-form"
                onSubmit={[Function]}
              >
                <Input
                  id="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                  value=""
                />
                <span
                  id="spanemail"
                />
                <Input
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
                <span
                  id="spanpassword"
                />
                <Input
                  className="submit"
                  id="submit"
                  name=""
                  placeholder=""
                  type="submit"
                  value="Login"
                />
                <Link
                  className="link"
                  to="/reset"
                >
                  Forgotten your password?
                </Link>
                <p
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Need an account?
                  <Link
                    className="link"
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </p>
              </form>
            </div>,
            "className": "d-flex justify-content-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="center"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-balance-scale fa-5x "
                  />
                  <h3
                    className="welcome__"
                  >
                    Please login
                  </h3>
                </div>,
                <form
                  className="signbox login"
                  id="login-form"
                  onSubmit={[Function]}
                >
                  <Input
                    id="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                    value=""
                  />
                  <span
                    id="spanemail"
                  />
                  <Input
                    id="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
                  />
                  <span
                    id="spanpassword"
                  />
                  <Input
                    className="submit"
                    id="submit"
                    name=""
                    placeholder=""
                    type="submit"
                    value="Login"
                  />
                  <Link
                    className="link"
                    to="/reset"
                  >
                    Forgotten your password?
                  </Link>
                  <p
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Need an account?
                    <Link
                      className="link"
                      to="/signup"
                    >
                      Sign Up
                    </Link>
                  </p>
                </form>,
              ],
              "className": "inner",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-balance-scale fa-5x "
                    />,
                    <h3
                      className="welcome__"
                    >
                      Please login
                    </h3>,
                  ],
                  "className": "center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-balance-scale fa-5x ",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Please login",
                      "className": "welcome__",
                    },
                    "ref": null,
                    "rendered": "Please login",
                    "type": "h3",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Input
                      id="email"
                      onChange={[Function]}
                      placeholder="Email"
                      required={true}
                      type="text"
                      value=""
                    />,
                    <span
                      id="spanemail"
                    />,
                    <Input
                      id="password"
                      onChange={[Function]}
                      placeholder="Password"
                      required={true}
                      type="password"
                      value=""
                    />,
                    <span
                      id="spanpassword"
                    />,
                    <Input
                      className="submit"
                      id="submit"
                      name=""
                      placeholder=""
                      type="submit"
                      value="Login"
                    />,
                    <Link
                      className="link"
                      to="/reset"
                    >
                      Forgotten your password?
                    </Link>,
                    <p
                      style={
                        Object {
                          "textAlign": "center",
                        }
                      }
                    >
                      Need an account?
                      <Link
                        className="link"
                        to="/signup"
                      >
                        Sign Up
                      </Link>
                    </p>,
                  ],
                  "className": "signbox login",
                  "id": "login-form",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "id": "email",
                      "onChange": [Function],
                      "placeholder": "Email",
                      "required": true,
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "spanemail",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "id": "password",
                      "onChange": [Function],
                      "placeholder": "Password",
                      "required": true,
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "spanpassword",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "submit",
                      "id": "submit",
                      "name": "",
                      "placeholder": "",
                      "type": "submit",
                      "value": "Login",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Forgotten your password?",
                      "className": "link",
                      "to": "/reset",
                    },
                    "ref": null,
                    "rendered": "Forgotten your password?",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Need an account?",
                        <Link
                          className="link"
                          to="/signup"
                        >
                          Sign Up
                        </Link>,
                      ],
                      "style": Object {
                        "textAlign": "center",
                      },
                    },
                    "ref": null,
                    "rendered": Array [
                      "Need an account?",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Sign Up",
                          "className": "link",
                          "to": "/signup",
                        },
                        "ref": null,
                        "rendered": "Sign Up",
                        "type": [Function],
                      },
                    ],
                    "type": "p",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        false,
        false,
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
